### PR TITLE
Create your own Enumerable

### PR DESCRIPTION
**Pull request by zewdu444**
# Create your own Enumerable

On this project we have been implemented the followings.

-  We had create a class `MyList` that has an instance variable `@list`.
- In `MyList` implement a method `#each` that yields successive members of `@list` and uses the `MyEnumerable` module.
- We had create a module `MyEnumerable` that implements the following methods 
  - `#all?`
  - `#any?`
  - `#filter`
 - We  create a different file for a class and module. 
 - We have been check  all linter error using `Rubocop`
 - We have been updated readme file

### CODE REVIEW

File: my_list.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+load 'my_enumerable.rb'
```
Comment (HAMMAS-SALEEM) [inline my_list.rb]: - [x] Kindly use `require_relative` instead of the `load` :+1: because require_relative is always used to import **my_enumerable.rb** file due to the following reasons :arrow_down:

**Performance:** load is a dynamic method, which means that it evaluates the specified file every time it's called. This can be slower than using require_relative, which loads the file only once and caches the result.

**Code readability:** Using load may make your code harder to read and understand. Since load can load files dynamically at runtime, it's not immediately clear to someone reading your code where the my_enumerable.rb file is coming from or what its contents are.

**Dependency management:** If you have many files in your project that depend on my_enumerable.rb, using load would make it harder to manage dependencies. You would need to ensure that the file is always available at runtime and in the correct location, whereas using require_relative would automatically handle this for you.

In summary, using load to load my_enumerable.rb is not ideal in this case because it could negatively impact performance, code readability, and dependency management. Therefore, it's recommended to use require_relative instead.

**Hint: You can see this [post](https://www.reddit.com/r/rails/comments/10nn0zb/the_difference_between_load_autoload_require_and/) to get more understanding about it :blush:**

File: README.md [hunk 2]
```diff
@@ -1,2 +1,128 @@
...
+- LinkedIn: [anthony-martin](https://www.linkedin.com/in/anthony-martin-8820a3117/)
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- FUTURE FEATURES -->
+
+## 🔭 Future Features <a name="future-features"></a>
+
+> We will be adding finding max and min values in the list.
+
```
Comment (HAMMAS-SALEEM) [inline README.md]: **[OPTIONAL]**

- [x] It would be nice if you add more features in this section such as below :point_down: to make your readme file more professional :+1:

* Improve Time Complexity
* Add UI

**Review by HAMMAS-SALEEM — CHANGES_REQUESTED** [review]
# Required Changes :recycle:
almost-there

## Hey @zewdu444 :wave:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following Gitflow
✔️ Following some requirements

Good job on implementing some of the requirements, there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: my_list.rb (hunk 1 above)
Comment (zewdu444) [reply my_list.rb]: @HAMMAS-SALEEM  thanks for the feedback and fixed based on your comment

File: README.md (hunk 2 above)
Comment (zewdu444) [reply README.md]: @HAMMAS-SALEEM more feature has been added

File: README.md [hunk 3]
```diff
@@ -90,6 +90,8 @@ To run the project, execute the following command in terminal:
 ## 🔭 Future Features <a name="future-features"></a>
 
 > We will be adding finding max and min values in the list.
+> Improve the code to be more readable.
+> Add more methods to the enumerable module.
```
Comment (HAMMAS-SALEEM) [inline README.md]: **[OPTIONAL]**

- [x] Your future features are looking like a paragraph :thinking: It would be nice if you put your Features like :arrow_down:

```sh
- We will be adding finding max and min values in the list.
- Improve the code to be more readable.
- Add more methods to the enumerable module.
```

**Review by HAMMAS-SALEEM — APPROVED** [review]
# Status: Approved :green_circle: :1st_place_medal: :100:
success

### Hey @zewdu444, @anthonymr 👋

Great work on implementing the requested changes. Your project is complete. :blush:
There is nothing else to say about your project. :+1:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following Gitflow
✔️ Following Requirements

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 3 above)
Comment (zewdu444) [reply README.md]: @HAMMAS-SALEEM  We also fixed this section.👍
